packages/webamp/js/components/FFTNullsoft.ts

```diff
@@ -0,0 +1,243 @@
+// The Web Audio API's FFT is bad, so this exists now!
+// Taken from https://github.com/WACUP/vis_classic/tree/master/FFTNullsoft
+
+export class FFT {
+    private m_samples_in: number;
+    private NFREQ: number;
+    private bitrevtable: number[] | null = null;
+    private envelope: Float32Array | null = null;
+    private equalize: Float32Array | null = null;
+    private temp1: Float32Array | null = null;
+    private temp2: Float32Array | null = null;
+    private cossintable: Float32Array[] | null = null;
+
+    constructor() {
+        this.m_samples_in = 0;
+        this.NFREQ = 0;
+    }
+
+    public init(samples_in: number, samples_out: number, bEqualize = 1, envelope_power = 1.0, mode = false): void {
```

Comment:
Can we simplify this code to hard code the variables that never vary? I think we only call this method once, and it's with hard coded values. Can we just inline those where they get used?

Reply:
I fail to understand what this really means, sorry

Reply:
Managed (or at least attempted to) address this.